### PR TITLE
4211-discoverable group

### PR DESCRIPTION
**Pull request by pkdash**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [x] Peer Code review and approval

### Positive Test Case
1. Create a group. Make the group discoverable. 
2. Logout of HydroShare
3. Click on COLLABORATE and then click on GROUPS.
4. In the Find Groups page, use the group search field to find the group you created. Verify you can see the group you created.
5. Login to HydroShare
6. Se the group that you created to private.
7. Logout of HydroShare
8. Click on COLLABORATE and then click on GROUPS.
9. In the Find Groups page, use the group search field to find the group you created. Verify you can't see the group you created

In addition to fixing the bug as described in #4211, this work also fixes several other issues as described in this document:

https://docs.google.com/spreadsheets/d/12rve0oe7Q3n0p7C3LpcGOdmpn7AinytslcbvUcuBCVw/edit#gid=0

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|
| NEW BUGS | 5 |
| NEW SECURITY HOTSPOTS | 0 |
| NEW CODE SMELLS | 0 |
| NEW VULNERABILITIES | 0 |
| NEW DUPLICATED LINES DENSITY | 62.1052631578947 |

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4419](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4419)

**Comment by Maurier** [comment]
@pkdash Is this ready for review?

**Comment by pkdash** [comment]
@Maurier I am waiting for @dtarb to test this work. Once he finishes testing and there are no issues, then it will be ready for review.

**Comment by pkdash** [comment]
@Maurier It's now ready for review.

File: theme/templates/pages/group.html [hunk 1]
```diff
@@ -58,6 +58,51 @@ <h2 class="group-title">{{ group.name }}</h2>
                 <h4 class="text-muted">{{ group.gaccess.purpose|linebreaks }}</h4>
 
                 <p>{{ group.gaccess.description|linebreaks }}</p>
+                {% if not profile_user.is_group_member and profile_user.is_authenticated %}
```
Comment (Maurier) [inline theme/templates/pages/group.html:61]: Making a note here to remember to update conditionals accordingly in work done in #4372

**Review by Maurier — APPROVED** [review]
Looks good. Verified test case.

**Pull request merged**